settings-boundary/EventSettings.tsx

```diff
@@ -0,0 +1,292 @@
+          )
+        }
+      })}
+    </View>
+  )
+}
+
+export const DurationSectionView = () => {
+  const { settings } = useUserSettings(settingsSelector("eventPresetDurations"))
+  const fontScale = useFontScale()
+  const sortedDurations = settings.eventPresetDurations.sort((a, b) => a - b)
+  return (
+    <SettingsSectionView title="Duration Presets">
+      <View style={styles.presetRowsGridContainer}>
+        {fontScale < FontScaleFactors.accessibility1 ? (
+          <>
+            <DurationCardRowView
+              sortedDurations={sortedDurations}
+              start={0}
+              end={3}
+            />
+            <DurationCardRowView
+              sortedDurations={sortedDurations}
+              start={3}
+              end={6}
+            />
+          </>
+        ) : (
+          <>
+            <DurationCardRowView
+              sortedDurations={sortedDurations}
+              start={0}
+              end={2}
+            />
+            <DurationCardRowView
+              sortedDurations={sortedDurations}
+              start={2}
+              end={4}
+            />
+            <DurationCardRowView
+              sortedDurations={sortedDurations}
+              start={4}
+              end={6}
+            />
+          </>
+        )}
+      </View>
+    </SettingsSectionView>
+  )
+}
+
+export type EventSettingsProps = {
+  style?: StyleProp<ViewStyle>
+  onLocationPresetTapped: (locationPreset: Placemark) => void
+  onDurationTapped: () => void
+}
+
+export const EventSettingsView = ({
+  style,
+  onLocationPresetTapped,
+  onDurationTapped
+}: EventSettingsProps) => {
+  return (
+    <SettingsScrollView style={style}>
+      <PresetSectionView
+        onLocationPresetTapped={onLocationPresetTapped}
+        onDurationTapped={onDurationTapped}
+      />
+    </SettingsScrollView>
+  )
+}
+
+export type EventDurationsProps = {
+  style?: StyleProp<ViewStyle>
+}
+
+export const EventDurationView = ({ style }: EventDurationsProps) => {
+  return (
+    <SettingsScrollView style={style}>
+      <DurationSectionView />
+    </SettingsScrollView>
+  )
+}
+
+type PresetSectionProps = {
+  onLocationPresetTapped: (locationPreset: Placemark) => void
+  onDurationTapped: () => void
+}
+
+const PresetSectionView = ({
+  onLocationPresetTapped,
+  onDurationTapped
+}: PresetSectionProps) => {
+  const { settings, update } = useUserSettings(
+    settingsSelector(
+      "eventPresetPlacemark",
+      "eventPresetShouldHideAfterStartDate"
+    )
+  )
+  return (
+    <SettingsCardSectionView
+      title="Presets"
+      subtitle="These presets will be populated when you create an event."
+    >
+      <SettingsNamedToggleView
+        name={"Hide After Start Date"}
+        description={
+          "The event will no longer be publicly visible to other users once it starts."
+        }
+        isOn={settings.eventPresetShouldHideAfterStartDate}
+        onIsOnChange={(eventPresetShouldHideAfterStartDate) =>
```

Comment:
unrelated to this pr but the name onIsOnChange is a confusing prop name, I'd maybe call it "enabled" and "onEnabledChange" or something similar